src/app/route/CategoryPage/CategoryPage.component.js

```diff
@@ -104,7 +104,10 @@ class CategoryPage extends Component {
 
         const totalPages = Math.ceil(totalItems / pageSize);
         const currentPage = parseInt(totalPages < pageFromUrl ? totalPages : pageFromUrl, 10);
-        const productsLoaded = Object.values(pages).reduce((accumulator, page) => accumulator + page.length, 0);
+
+        const pagesValues = Object.values(pages);
+        const productsLoaded = pagesValues.length
+            ? pagesValues.reduce((accumulator, page) => accumulator + page.length, 0) : 0;
```

Comment:
Optimise this into:

```js
const productsLoaded = pagesValues.reduce((accumulator, page) => accumulator + page.length, 0);
```

Because reduce returns its initial value if array length is 0.